File: src/main/java/com/example/quanlynhansu/service/AttendanceService.java
package com.example.quanlynhansu.service;

import com.example.quanlynhansu.dto.AttendanceDTO;
import com.example.quanlynhansu.dto.CheckInRequest;
import com.example.quanlynhansu.entity.Attendance;
import java.util.List;

public interface AttendanceService {

    /**
     * Xử lý hành động Check-in (Vào ca)
     * @param userId: ID nhân viên
     * @param request: Thông tin toạ độ, ghi chú
     * @return Bản ghi chấm công vừa tạo
     */
    Attendance checkIn(Long userId, CheckInRequest request);

    /**
     * Xử lý hành động Check-out (Tan ca)
     * @param userId: ID nhân viên
     * @return Bản ghi chấm công đã cập nhật giờ về
     */
    Attendance checkOut(Long userId);
   List<Attendance> getTodayList(Long userId);

    // --- THÊM DÒNG NÀY ĐỂ KHỚP VỚI CODE BẠN VỪA LÀM ---
    List<AttendanceDTO> getAttendanceHistory(Long userId, int month, int year);
    String processFaceCheckIn(Long userId);
}

File: forntend_quanlynhansu/src/services/AttendanceService.js
import BaseApi from "./BaseApi";

const API_URL = "/attendance";

const checkIn = (userId, data) => {
    return BaseApi.post(`${API_URL}/check-in?userId=${userId}`, data);
};

const checkOut = (userId) => {
    return BaseApi.post(`${API_URL}/check-out?userId=${userId}`);
};

const getToday = (userId) => {
    return BaseApi.get(`${API_URL}/today?userId=${userId}`);
};

const getHistory = (userId, month, year) => {
    return BaseApi.get(`${API_URL}/history?userId=${userId}&month=${month}&year=${year}`);
};

export default { checkIn, checkOut, getToday, getHistory };